site/src/pages/WorkspaceSchedulePage/WorkspaceSchedulePage.tsx

```diff
@@ -172,6 +172,7 @@ export const WorkspaceSchedulePage: React.FC = () => {
             autoStart: formValuesToAutoStartRequest(values),
             ttl: formValuesToTTLRequest(values),
           })
+          return Promise.resolve()
```

Comment:
**Review**:

Useful whenever you're returning a synchronous `onSubmit`:

https://formik.org/docs/api/formik#onsubmit-values-values-formikbag-formikbag--void--promiseany

It looks a little janky (I know), but is absolutely necessary to avoid infini-spinning submit button if submissions fails the first time.

Reply:
I was avoiding using formik's `isSubmitting` (for instance, to determine whether to disable things), but are you saying that formik internally uses it so we still need this workaround?

Reply:
Yeah, I see it as letting XState be in charge of knowing what state we're in instead of Formik.